### PR TITLE
Agent exit code improvements

### PR DESCRIPTION
**Pull request by mprimi**

### CODE REVIEW

File: genie-agent/src/main/java/com/netflix/genie/agent/cli/ExitCode.java [hunk 1]
```diff
@@ -49,7 +49,12 @@
     /**
      * The selected agent command failed to execute.
      */
-    EXEC_FAIL(104, "Command execution error");
+    EXEC_FAIL(104, "Command execution error"),
```
[the remark this answers is not in the document]
Comment (mprimi) [reply genie-agent/src/main/java/com/netflix/genie/agent/cli/ExitCode.java]: The tickets says that that's an **example** of convention for exit codes

**Comment by mprimi** [comment]
@tgianos at the moment, this would produce an exit code `EXEC_FAIL` (`Command execution error`).
However this approach leaves the door open to more fine-grained exit codes later (for example we may distinguish agent errors such as server unreachable from job failing to execute).
At the moment, the request was just
 - Zero for successful job
 - Non-zero for job failed
 - Non-zero and different than job failed for job killed

**Comment by sumitnetflix** [comment]
@mprimi I see that `GenieAgentRunner` calls `command.run()` here - https://github.com/mprimi/genie/blob/GENIE-313-agent-exit-code/genie-agent/src/main/java/com/netflix/genie/agent/cli/GenieAgentRunnner.java#L98. When `ExecCommand` throws `RuntimeException`, I dont see how this gets mapped to EXEC_FAIL. Am I seeing something wrong?

**Comment by mprimi** [comment]
Happens on the line above the one you linked.

(also what's the reason for sending a one-off comment linking to my fork, rather than discussing inline in the PR?)

**Pull request merged**